hail/src/main/scala/is/hail/expr/ir/lowering/LowerTableIR.scala

```diff
@@ -506,6 +506,58 @@ object LowerTableIR {
             sorted
           }
 
+        case TableLeftJoinRightDistinct(left, right, root) =>
+          require(right.typ.keyType isPrefixOf left.typ.keyType)
+
+          val commonKeyLength = right.typ.keyType.size
+          val loweredLeft = lower(left)
+          val leftKeyToRightKeyMap = left.typ.keyType.fieldNames.zip(right.typ.keyType.fieldNames).toMap
+          val newRightPartitioner = loweredLeft.partitioner.strictify.coarsen(commonKeyLength)
+            .rename(leftKeyToRightKeyMap)
```

Comment:
I don't think the rename is necessary. Partitions should ignore field names completely. If that's not true, I'd like to know about it.

Reply:
I think that the names sort of matter, in that it's pretty common throughout the lowering code to get the key fields by doing `lowered.partitioner.kType`. Let me investigate

Reply:
Ah, yeah, you're probably right.

Reply:
This is the reason for having this rename:

https://github.com/hail-is/hail/blob/master/hail/src/main/scala/is/hail/expr/ir/lowering/LowerTableIR.scala#L157

Shuffling uses the key names for the partitioner to `SelectFields`